src/guacscanner/guacscanner.py

```diff
@@ -0,0 +1,758 @@
+    {entity_id_field}, {connection_id_field}, {permission_field})
+    VALUES (%s, %s, %s);"""
+).format(
+    table=sql.Identifier("guacamole_connection_permission"),
+    entity_id_field=sql.Identifier("entity_id"),
+    connection_id_field=sql.Identifier("connection_id"),
+    permission_field=sql.Identifier("permission"),
+)
+DELETE_CONNECTION_PERMISSIONS_QUERY = sql.SQL(
+    """DELETE FROM {table} WHERE {connection_id_field} = %s;"""
+).format(
+    table=sql.Identifier("guacamole_connection_permission"),
+    connection_id_field=sql.Identifier("connection_id"),
+)
+
+
+def entity_exists(db_connection, entity_name, entity_type):
+    """Return a boolean indicating whether an entity with the specified name and type exists."""
+    with db_connection.cursor() as cursor:
+        logging.debug(
+            "Checking to see if an entity named %s of type %s exists in the database.",
+            entity_name,
+            entity_type,
+        )
+        cursor.execute(ENTITY_COUNT_QUERY, (entity_name, entity_type))
+        count = cursor.fetchone()["count"]
+        if count != 0:
+            logging.debug(
+                "An entity named %s of type %s exists in the database.",
+                entity_name,
+                entity_type,
+            )
+        else:
+            logging.debug(
+                "No entity named %s of type %s exists in the database.",
+                entity_name,
+                entity_type,
+            )
+
+        return count != 0
+
+
+def get_entity_id(db_connection, entity_name, entity_type):
+    """Return the ID corresponding to the entity with the specified name and type."""
+    logging.debug("Looking for entity ID for %s of type %s.", entity_name, entity_type)
+    with db_connection.cursor() as cursor:
+        logging.debug(
+            "Checking to see if any entity named %s of type %s exists in the database.",
+            entity_name,
+            entity_type,
+        )
+        cursor.execute(ENTITY_ID_QUERY, (entity_name, entity_type))
+
+        # Note that we are assuming there is only a single match.
+        return cursor.fetchone()["entity_id"]
+
+
+def add_user(db_connection, username):
+    """Add a user, returning its corresponding entity ID."""
+    logging.debug("Adding user entry for %s.", username)
+    entity_id = None
+    with db_connection.cursor() as cursor:
+        cursor.execute(
+            INSERT_ENTITY_QUERY,
+            (
+                username,
+                "USER",
+            ),
+        )
+        entity_id = cursor.fetchone()["entity_id"]
+        cursor.execute(
+            INSERT_USER_QUERY,
+            (
+                entity_id,
+                # guacadmin
+                bytes.fromhex(
+                    "CA458A7D494E3BE824F5E1E175A1556C0F8EEF2C2D7DF3633BEC4A29C4411960"
+                ),
+                bytes.fromhex(
+                    "FE24ADC5E11E2B25288D1704ABE67A79E342ECC26064CE69C5B3177795A82264"
+                ),
```

Comment:
My main problem with generating random passwords is that I can't for the life of me obtain the password hash above from the password `guacadmin` and the given salt, despite spending hours with [these instructions](https://guacamole.apache.org/doc/gug/jdbc-auth.html#jdbc-auth-schema-users).  I guess that would be ultra secure, since I'd be setting passwords that even I don't know.

If I could generate the example password hash from the known password and salt then I would happily create random passwords.  I beg someone to show me that I am a fool and have misread the instructions linked to in the previous paragraph before I sink further into madness.

Given that we don't use the passwords, are you OK with me adding the above two paragraphs in a comment and creating an issue to create random passwords in the future?

Reply:
@mcdonnnj has shown me the light WRT generating the passwords.  I will change the code to generate a random password.

Reply:
Please have a look at commit e595bb3.